lib/coverage.js

```diff
@@ -269,19 +269,41 @@ internals.instrument = function (filename) {
 
     let skipStart = 0;
     let segmentSkip = false;
+
+    // In addition to maintaining the current coverage bypass state, support
+    // comments that push that state to a stack, allowing transpiled code to
+    // skip coverage without interfering with a user-defined bypass block.
+    const skipStack = [];
     tree.comments.forEach((comment) => {
 
-        const directive = comment.value.match(/^\s*\$lab\:coverage\:(off|on)\$\s*$/);
+        const directive = comment.value.match(/^\s*\$lab\:coverage\:(off|on|push|pop)\$\s*$/);
         if (directive) {
-            const skip = directive[1] !== 'on';
-            if (skip !== segmentSkip) {
-                segmentSkip = skip;
-                if (skip) {
-                    skipStart = comment.range[1];
+            const command = directive[1];
+            if (command === 'push') {
+                skipStack.push(segmentSkip);
+            }
+            else {
+                let newSkipState;
+                if (command === 'pop') {
+                    if (!skipStack.length) {
+                        throw new Error('unable to pop coverage bypass stack');
+                    }
+
+                    newSkipState = skipStack.pop();
                 }
                 else {
-                    for (let i = skipStart; i < comment.range[0]; ++i) {
-                        bypass[i] = true;
+                    newSkipState = command !== 'on';
+                }
+
+                if (newSkipState !== segmentSkip) {
+                    segmentSkip = newSkipState;
+                    if (newSkipState) {
+                        skipStart = comment.range[1];
+                    }
+                    else {
+                        for (let i = skipStart; i < comment.range[0]; ++i) {
```

Comment:
it's just occurring to me that a `$lab:coverage:off$` comment without a subsequent `on` comment might result in no bypass - is this intended behavior, or just a property of how this was originally structured?